### PR TITLE
Fix test_configdrive.py and test_nuage_configdrive

### PR DESCRIPTION
**Pull request by fmaximus**
## Description

Some fixes to the marvin tests were still ongoing, when PR #2566 was merged.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

Ran regression on Nuage setup.

## Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [x] I have added tests to cover my changes.
- [x] All relevant new and existing integration tests have passed.
- [x] A full integration testsuite with all test that can run on my environment has passed.

### CODE REVIEW

**Comment by borisstoyanov** [comment]
@fmaximus thanks for the fixes, let me see how this behaves in trillian and will get back to you!

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
Fixes: #2671 
doesn't it!?

**Comment by rhtyd** [comment]
@blueorangutan help

**Comment by rhtyd** [comment]
@blueorangutan test centos7 kvm-centos7 component/test_configdrive.py

File: test/integration/component/test_configdrive.py [hunk 1]
```diff
@@ -513,17 +535,22 @@ def update_provider_state(self, new_state):
...
         return orig_state
 
+    def _get_test_data(self, key):
+        return self.test_data[key]
+
+    def get_configdrive_provider(self):
+        return NetworkServiceProvider.list(
+            self.api_client,
+            name="ConfigDrive")[0]
+
```
Comment (GabrielBrascher) [inline test/integration/component/test_configdrive.py]: Shouldn't it have a line `physicalnetworkid=self.vsp_physical_network.id)[0]`? I am following the same logic of the test `get_configdrive_provider` on `test_nuage_configdrive.py`?

Comment (fmaximus) [reply test/integration/component/test_configdrive.py]: Native test can just take the first one, as there are no multiple physical networks which might have a ConfigDrive provider, as is the case in a Nuage setup

Comment (GabrielBrascher) [reply test/integration/component/test_configdrive.py]: Got it, thanks.

**Comment by DaanHoogland** [comment]
since this is only test code I am taking the liberty to merge

**Comment by fmaximus** [comment]
As password reset now requires the VM's to be stopped (PR #2651),
both test needs to be adapted again.

**Pull request merged**